train.py

```diff
@@ -252,8 +256,12 @@ def main():
 
             if step % 50 == 0:
                 save(saver, sess, logdir, step)
+                last_save_step = step
 
     finally:
+        if step > last_save_step:
+            print()
```

Comment:
I see, that makes a lot of sense. In that case, could you add an `except KeyboardInterrupt:` block before `finally`, and do it there together with an explanatory comment (`# Introduce a line break after ^C`, or something like that)? It would make it much clearer, and it's a very neat trick; having it well documented will help other people (like me) learn it and use it elsewhere.

This is only relevant, though, if you are displaying stdout on the terminal, as opposed to redirecting and writing to a log file.

I'm also wondering if there's a more explicit way of printing an empty line. Maybe `print(end='\n')`? It feels like too much though, I think the current implementation is cleaner.